### ISSUE TITLE
Problem using fo-dicom with Xamarin

### ISSUE THREAD

**Issue by GeovaneW96** [fo-dicom/fo-dicom#1169]
**Expected behavior**
Successfully use fo-dicom in a Xamarin.Mac project.

**Actual behavior**
I get an error saying "Could not find `PresentationCore` referenced by assembly `Dicom.Core`" 

**Steps to reproduce the behavior**
I have a solution with 2 projects.
One project is a Library and targets .NET Framework 4.5.2, this project uses fo-dicom
And another project that is an Executable, targets Xamarin.Mac Full .NET Framework 4.5.2 and references the Library.
The build is done just fine, but when executing the exception "Could not find `PresentationCore` referenced by assembly `Dicom.Core`" occurs from the executable project.

**fo-dicom version and OS/platform**
fo-dicom 4.0.7
macOS Sierra 10.12.16

**Comment by gofal** [comment]
yes, building in various platforms and therefore referencing some dependencies always caused issues, some of then hard to solve. This was because of the structure of having various builds for various platforms.

This should be solved with version 5 of fo-dicom. Please see here: https://github.com/fo-dicom/fo-dicom/wiki/Upgrade-from-version-4-to-version-5
This version now only has one assembly compiled in netstandard2.0. There should not be any referecing-issues any more with that.

Please give it a try and give feedback! the goal is to make version 5 work smooth rather ran fixing structural issues in version 4.

**Comment by GeovaneW96** [comment]
I had to upgrade all my projects to .NET Framework 4.6.1 to support net standard 2.0 and it was a really big pain.

Anyway, when I finally made it, I get the exception:

"Exception" Decoding dataset with transfer syntax: JPEG 2000 Image Compression (Lossless Only) is not supported

So, how am I supposed to convert  Jpeg2K to any other uncompressed transfer syntax?

**Comment by gofal** [comment]
The codec, that was delivered with fo-dicom by joining various other open source projects, only supported windows desktop. See this table of supported codecs in fo-dicom 4: https://github.com/fo-dicom/fo-dicom/wiki/Supported-Transfer-Syntaxes. You will notice that with xamarin (android, ios, ..) jpeg2k was only supported for 8 bit images, which is good in computer vision, but not in radiology where images have 10, 12 or 16 bit.
We are glad, that EfferentHealth, an US company, adopted the codec to not only work on windows, but also on other operating systems. So please reference the nuget package `fo-dicom.Codecs` by Efferent-Health. And then somewhere in your initialization step register this codec by calling:

```
using FellowOakDicom;
using FellowOakDicom.Imaging.NativeCodec;

//....

   new DicomSetupBuilder()
       .RegisterServices(s => s.AddTranscoderManager<NativeTranscoderManager>())
       .Build();
```

**Comment by GeovaneW96** [comment]
Thank you.

I did as you said and now I'm getting the following Exception:

{System.DllNotFoundException: Dicom.Native-macOS.dylib
  at (wrapper managed-to-native) FellowOakDicom.Native.Codec.DicomJpeg2000NativeCodec.Memset_MacOS(void*,int,uint)
  at FellowOakDicom.Native.Codec.DicomJpeg2000NativeCodec.Decode (FellowOakDicom.Imaging.DicomPixelData oldPixelData, FellowOakDicom.Imaging.DicomPixelData newPixelData, FellowOakDicom.Imaging.Codec.DicomCodecParams parameters) [0x001f4] in <97b4f990857b484e80caba658c77b536>:0 
  at FellowOakDicom.Imaging.Codec.DicomTranscoder.Decode (FellowOakDicom.DicomDataset oldDataset, FellowOakDicom.DicomTransferSyntax outSyntax, FellowOakDicom.Imaging.Codec.IDicomCodec codec, FellowOakDicom.Imaging.Codec.DicomCodecParams parameters) [0x00037] in <916ea9232b204d3aaaf1bc98c7ad4299>:0 
  at FellowOakDicom.Imaging.Codec.DicomTranscoder.Transcode (FellowOakDicom.DicomDataset dataset) [0x000ff] in <916ea9232b204d3aaaf1bc98c7ad4299>:0 
  at FellowOakDicom.Imaging.Codec.DicomTranscoder.Transcode (FellowOakDicom.DicomFile file) [0x00053] in <916ea9232b204d3aaaf1bc98c7ad4299>:0 }

**Comment by gofal** [comment]
oh no, sorry I cannot help you, because I have no experience with mac. 
efferent wrote on their homepage that the codec supports mac:
> MacOS (tested with macOS High Sierra Version 10.13)
are you using another os version?
